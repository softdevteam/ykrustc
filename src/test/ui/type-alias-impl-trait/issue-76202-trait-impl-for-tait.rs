// Regression test for issue #76202
// Tests that we don't ICE when we have a trait impl on a TAIT.

// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete

trait Dummy {}
impl Dummy for () {}

type F = impl Dummy;
fn f() -> F {}

trait Test {
    fn test(self);
}

impl Test for F { //~ ERROR cannot implement trait
    fn test(self) {}
}

fn main() {
    let x: F = f();
    x.test();
}
