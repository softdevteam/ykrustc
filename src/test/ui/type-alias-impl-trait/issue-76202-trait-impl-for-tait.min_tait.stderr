error: cannot implement trait on type alias impl trait
  --> $DIR/issue-76202-trait-impl-for-tait.rs:19:1
   |
LL | impl Test for F {
   | ^^^^^^^^^^^^^^^
   |
note: type alias impl trait defined here
  --> $DIR/issue-76202-trait-impl-for-tait.rs:12:10
   |
LL | type F = impl Dummy;
   |          ^^^^^^^^^^

error: aborting due to previous error

